activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -84,7 +84,8 @@ def initialize(secret, *signature_key_or_options)
 
     # Encrypt and sign a message. We need to sign the message in order to avoid
     # padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
-    def encrypt_and_sign(value)
+    def encrypt_and_sign(value, meta_data = {})
+      value = { value: value, meta_data: meta_data } unless meta_data.empty?
```

Comment:
A note about backwardscompatibility: the tests still output the same encrypted messages now, but should they fail that means we've broken backwardscompatibility. So keep an eye out for that 😊